cyclonedx-bom/src/external_models/uri.rs

```diff
@@ -16,16 +16,32 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-use std::str::FromStr;
+use std::{convert::TryFrom, str::FromStr};
 
 use packageurl::PackageUrl;
+use thiserror::Error;
 
 use crate::validation::{
     FailureReason, Validate, ValidationContext, ValidationError, ValidationResult,
 };
 
-#[derive(Debug, PartialEq, Eq)]
-pub struct Purl(pub(crate) Uri);
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Purl(pub(crate) String);
+
+impl Purl {
+    pub fn new(name: &str, version: &str, package_type: &str) -> Result<Purl, UriError> {
```

Comment:
Style preference: I'd probably have `package_type` first, since that's the order it shows up in a purl.